doc/man1/flux-alloc.rst

```diff
@@ -13,8 +13,40 @@ SYNOPSIS
 DESCRIPTION
 ===========
 
-flux-alloc(1) interactively launches a command as the initial program of a
-new Flux instance.  If no *COMMAND* is specified, a shell is spawned.
+**flux-alloc** runs a Flux subinstance with *COMMAND* as the initial program.
+Once resource are allocated, *COMMAND* executes on the first node of the
+allocation with any free arguments supplied as *COMMAND* arguments.  When
+*COMMAND* exits, the Flux subinstance exits, resources are released to the
+enclosing Flux instance, and **flux-alloc** returns.
+
+If no *COMMAND* is specified, an interactive shell is spawned as the initial
+program, and the subinstance runs until the shell is exited.
+
+If the *--bg* option is specified, the subinstance runs without an initial
+program.  **flux-alloc** prints the job ID and returns as soon as the
+subinstance is ready to accept jobs.  The subinstance runs until it exceeds
+its time limit, is canceled, or is shut down with :man1:`flux-shutdown`.
+
+Flux commands that are run from the subinstance (e.g. from the interactive
+shell) refer to the subinstance. For example, :man1:`flux-run` would launch
+work there.  A Flux command run from the subinstance can be forced to refer
+to the enclosing instance by supplying the :man1:`flux` --parent option.
+
+Flux commands outside of the subinstance refer to their enclosing instance,
+often a system instance. flux-proxy(1) establishes a connection to a running
+subinstance by job ID, then spawns a shell in which Flux commands refer to
```

Comment:
not  necessary to address in the PR, but we alternate use of `job ID` and `jobid` to refer to the jobid throughout the docs. I think `jobid` is used far more and has the benefit of matching parameters used in APIs and relevant broker attributes etc, though I don't necessarily think "job ID" is wrong. Just bringing it up as something to think about.

Reply:
Makes sense.  I can switch to `jobid` here at least.